### PR TITLE
Redis ping check

### LINKED ISSUE #188: blnk can start without redis
Redis is required for blnk, but it's possible to start server without it.

**To Reproduce**
1. start blnk without redids
2. create transaction
3. receive 400 status code with message cannot dial redis

**Expected behavior**
blnk shouldn't start without redis and status code shoudn't be 400

**Screenshots**

Image

### PR DESCRIPTION
**Pull request by RyanCarlisle**
Fixes #188 

tr1

### CODE REVIEW

File: internal/redis-db/redis_test.go [hunk 1]
```diff
@@ -71,12 +71,12 @@ func TestNewRedisClient(t *testing.T) {
 		{
 			name:      "single address",
 			addresses: []string{"localhost:6379"},
-			wantErr:   false,
```
Comment (jerry-enebeli) [inline internal/redis-db/redis_test.go:74]: I think `wantErr` should be `false`, given that the URL is correct, present, and uses a standalone(single URL).

It'd only be `true` when the Redis instance is not running. 

We can extend the test to add a wrong port or host and expect an error(the ping should throw an error)

**Review by jerry-enebeli — CHANGES_REQUESTED** [review]
Thank you for the PR @RyanCarlisle

File: internal/redis-db/redis_test.go (hunk 1 above)
Comment (RyanCarlisle) [reply internal/redis-db/redis_test.go:74]: The issue here is if Redis is not set up already, and you run the test case, the test case itself will panic as it will be trying to assert on NoError. This is the reason why I had made wantErr `true`. Could you let me know what is to be done instead in such cases?

Comment (jerry-enebeli) [reply internal/redis-db/redis_test.go:74]:
```suggestion
		{
			name:      "single address with wrong port",
			addresses: []string{"localhost:63749"},
			wantErr:   true,
		},
```

We can add one more test to simulate Redis not being up or ready (using an incorrect port). This should cover your edge case

**Comment by artempanko** [comment]
> Hi @artempanko , could you elaborate on what it is that is required here?

Just add restart on failure, like for PostgreSQL and typesense
